### ISSUE TITLE
Miner got slashed immediately after its first reward claim tx

### ISSUE THREAD

**Issue by h4x3rotab** [Phala-Network/phala-blockchain#186]
Here is a sample, and it affect everyone on the network:

- stash: 45psUjc7jD6Nj3y8o27dxif71CGnTE7YgVTd17jkFqWWhCf5
- controller: 45yCuH7m4fvzCyYNXj3g9DdyF6iay6GfLk9Q3xPaWQgCRq74

Timeline on poc3:

1. Registered at 1820543
2. Started mining at 1820640
3. Claim the first reward at  1821428 (sequence 0)
4. Slash event 1821457-2: was triggered, reporting the reward challenge @1821425 is not responded:
    <https://phala.subscan.io/extrinsic/0xa295d17177fe2056dd3bb01cfe1700b3d06b23f538b79684967de106da60c69f?event=1821457-2>
5. Claim the second reward failed at 1821587 (failed with BadWorkerMessageSequence where sequence = 1)

**Comment by h4x3rotab** [comment]
With `scripts/js/b186-check-miner-sequence.js`, we can find that the `lastWorkerActivity` is always zero. It turned out the controller address was wrongly used as the key, but when reading from it, the key is stash.

```js
{
  controller: '45yCuH7m4fvzCyYNXj3g9DdyF6iay6GfLk9Q3xPaWQgCRq74',
  seqAfterReg: 0,
  miningStatusBegin: {
    machineId: '0x78dce511327623abe5055dce93bffdfa',
    pubkey: '0x02656fd4e2cddd2273a95b5d0b0937b7beae429b873122ec345c62753c7d8abae6',
    lastUpdated: '1,618,112,820,000',
    state: { MiningPending: null },
    score: { overallScore: '1,280', features: [ '16', '2' ] }
  },
  activityBegin: 0,
  seqAfterHit1: 1,
  miningStatusHit1: {
    machineId: '0x78dce511327623abe5055dce93bffdfa',
    pubkey: '0x02656fd4e2cddd2273a95b5d0b0937b7beae429b873122ec345c62753c7d8abae6',
    lastUpdated: '1,618,112,820,000',
    state: { Mining: '1,821,000' },
    score: { overallScore: '1,280', features: [ '16', '2' ] }
  },
  activityHit1: 0,
  seqAfterHit2: 0,
  miningStatusHit2Before: {
    machineId: '0x78dce511327623abe5055dce93bffdfa',
    pubkey: '0x02656fd4e2cddd2273a95b5d0b0937b7beae429b873122ec345c62753c7d8abae6',
    lastUpdated: '1,618,118,712,006',
    state: { Free: null },
    score: { overallScore: '1,280', features: [ '16', '2' ] }
  },
  miningStatusHit2: {
    machineId: '0x78dce511327623abe5055dce93bffdfa',
    pubkey: '0x02656fd4e2cddd2273a95b5d0b0937b7beae429b873122ec345c62753c7d8abae6',
    lastUpdated: '1,618,118,712,006',
    state: { Free: null },
    score: { overallScore: '1,280', features: [ '16', '2' ] }
  },
  activityHit2: 0
}
```

**Comment by h4x3rotab** [comment]
Fixed by 7e8acb09aaa731add85111e93629694930579ab4

We will need to backport this to poc3 and do an onchain upgrade to rescue the testnet

**Comment by h4x3rotab** [comment]
The PoC3 runtime has been upgraded to v32.
